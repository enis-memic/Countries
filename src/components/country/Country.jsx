import './country.css';
import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { BsFillArrowRightCircleFill } from 'react-icons/bs';
import {
  searchByRegion,
  showAllCountries,
} from '../../redux/countries/countriesAction';

const Country = () => {
  const {
    countriesData, loading, success, error, region, searchTerm,
  } = useSelector((state) => state.country);

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(showAllCountries());

    if (region) {
      dispatch(searchByRegion(region));
    }

    if (error) {
      dispatch(error);
    }
  }, [dispatch, error, success, region]);

  const data = countriesData.filter((item) => item.name.common.toLowerCase().includes(searchTerm));

  return (
    <section className="country-container">
      {loading ? (
        <h1>Loading</h1>
      ) : (
        data.length > 0
        && data.map((item, index) => (
          <Link className="country-card" key={index} to={`/${item.cioc}`}>
            <p className="more">
              {' '}
              <BsFillArrowRightCircleFill className="icon-more" />
            </p>
            <img
              src={item.flags.png}
              alt={item.flags.alt}
              className="country-image"
            />
            <div className="country-content">
              <h3>
                {item.name.common}
                {' '}
              </h3>
              <p>
                {' '}
                <span>{item.population}</span>
              </p>
            </div>
          </Link>
        ))
      )}
    </section>
  );
};

export default Country;
