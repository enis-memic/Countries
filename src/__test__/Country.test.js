import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Country from '../components/country/Country';
import { store } from '../redux/store';

describe('Navbar rendering', () => {
  test('Navbar rendering', () => {
    const render = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Country />
        </BrowserRouter>
      </Provider>
    );
    expect(render).toMatchSnapshot();
  });
});
