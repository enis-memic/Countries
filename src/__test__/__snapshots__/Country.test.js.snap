// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar rendering Navbar rendering 1`] = `
<section
  className="country-container"
/>
`;
